eventmesh-connectors/eventmesh-connector-http/src/main/java/org/apache/eventmesh/connector/http/sink/handle/WebhookHttpSinkHandler.java

```diff
@@ -129,66 +194,118 @@ public void start() {
+            HttpExportMetadataBuilder builder = HttpExportMetadata.builder()
+                .url(url.toString())
+                .receivedTime(LocalDateTime.now())
+                .retriedBy(null)
+                .uuid(UUID.randomUUID().toString())
+                .retryNum(0);
+
+            if (arr.succeeded()) {
+                HttpResponse<Buffer> response = arr.result();
+                builder.code(response.statusCode())
+                    .message(response.statusMessage());
+            } else {
+                builder.code(-1)
+                    .message(arr.cause().getMessage());
             }
-            // If the queue is full, remove the oldest element
-            if (receivedDataQueue.size() == Integer.MAX_VALUE) {
-                Object removedData = receivedDataQueue.poll();
-                if (log.isDebugEnabled()) {
-                    log.debug("The queue is full, remove the oldest element: {}", removedData);
-                } else {
-                    log.info("The queue is full, remove the oldest element");
-                }
+            // create ExportRecord
+            HttpExportRecord exportRecord = new HttpExportRecord(builder.build(), arr.succeeded() ? arr.result().bodyAsString() : null);
+            // add the data to the queue
+            addDataToQueue(exportRecord);
+        });
+    }
+
+
+    /**
+     * Adds the received data to the queue.
+     *
+     * @param exportRecord the received data to add to the queue
+     */
+    public void addDataToQueue(HttpExportRecord exportRecord) {
+        // If the current queue size is greater than or equal to the maximum queue size, remove the oldest element
+        if (currentQueueSize.get() >= maxQueueSize) {
+            Object removedData = receivedDataQueue.poll();
+            if (log.isDebugEnabled()) {
+                log.debug("The queue is full, remove the oldest element: {}", removedData);
+            } else {
+                log.info("The queue is full, remove the oldest element");
             }
-            // Try to put the received data into the queue
-            if (receivedDataQueue.offer(receivedData)) {
-                if (log.isDebugEnabled()) {
-                    log.debug("Successfully put the received data into the queue: {}", receivedData);
-                } else {
-                    log.info("Successfully put the received data into the queue");
-                }
+            currentQueueSize.decrementAndGet();
+        }
+        // Try to put the received data into the queue
+        if (receivedDataQueue.offer(exportRecord)) {
+            currentQueueSize.incrementAndGet();
+            if (log.isDebugEnabled()) {
+                log.debug("Successfully put the received data into the queue: {}", exportRecord);
             } else {
-                log.error("Failed to put the received data into the queue: {}", receivedData);
+                log.info("Successfully put the received data into the queue");
```

Comment:
```java
log.debug("Successfully put the received data into the queue: {}", exportRecord);
```

This will be enough here. Putting something into the queue is not an important thing to log.

Reply:
OK. I will make the modification.

Reply:
Resolved at https://github.com/apache/eventmesh/pull/4837/commits/4af7455c9b09d4a1771fa7580b7406a61160ea74